id: allow_failure
namespace: example

description: This flow will allow a failure of a task (imagine a flaky unit test) and will continue processing the last task, leaving the execution in a `WARNING` state.

tasks:
  - id: first
    type: io.kestra.core.tasks.debugs.Return
    format: "{{task.id}} > {{taskrun.startDate}}"

  - id: allow_failure
    type: io.kestra.plugin.scripts.shell.Commands
    runner: PROCESS
    allowFailure: true
    commands:
      - exit 1

  - id: last
    type: io.kestra.core.tasks.debugs.Return
    format: "{{task.id}} > {{taskrun.startDate}}"